### PR TITLE
bug(auth):fix auth vulnerabilities

### PR DESCRIPTION
**Pull request by RusimbiPatrick**
### What does this PR do?

- Configure the app to get and use a token from headers
- Update Swagger Docs due to the changes
- Refactor tests due to the changes
- Emphasize strong passwords
- Add HTTP request rate limit for one IP Address to prevent DDoS Attacks

### How should this be manually tested?

- Using the Heroku link test the whole app in `Swagger Docs` or `POSTMAN`

### Implementation plan

[Implementation plan name](link to the implementation plan document)

### Stories associated with PR

[# 171870345(https://www.pivotaltracker.com/story/show/171967419)

### Screenshots if any
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,6 +45,7 @@
     "errorhandler": "^1.5.0",
     "express": "^4.16.3",
     "express-jwt": "^5.3.1",
+    "express-rate-limit": "^5.1.1",
```
Comment (blaise82) [inline package.json:48]: what does package-lock.json do?

File: src/controllers/userController.js [hunk 2]
```diff
@@ -37,7 +36,7 @@ class userController {
       const Token = encode({
```
Comment (blaise82) [inline src/controllers/userController.js]: can encode the whole user object(excluding the password) in the token and make sure that it's returned too (return the token)

File: src/controllers/userController.js [hunk 3]
```diff
@@ -55,26 +54,26 @@ class userController {
...
+  //       status: 403,
+  //       message: setLanguage(preferLang).__('Pleaselog'),
+  //     });
+  //   }
+  //   localStorage.removeItem('Token', 0);
+  //   return res.status(200).json({
+  //     status: 200,
+  //     message: setLanguage(preferLang).__('outSuccessfully'),
+  //   });
+  // }
```
Comment (blaise82) [inline src/controllers/userController.js]: just delete it

File: src/tests/mock/tokens.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+export const octopusbn = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6Im9jdG9wdXNibkBnbWFpbC5jb20iLCJwcmVmZXJlZExhbmciOiJlbiIsImlhdCI6MTU4NTM2MTI0M30.7R1fEtcHu-LJI9WseFsClQWV4nIk-MCGl_y07-dM9D0"
+export const testManager = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6Im5lZWRzLmdyaWRAZ21haWwuY29tIiwicHJlZmVyZWRMYW5nIjoiZW4iLCJpYXQiOjE1ODUzNjEyNzV9.Fxa-VmiWy8IieuAJTxQT3knApZPrhc_JVHWXrtGTctg"
+export const rusimbipatrick = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InJ1c2ltYmlwYXRyaWNrQG91dGxvb2suY29tIiwicHJlZmVyZWRMYW5nIjoiZnIiLCJpYXQiOjE1ODUzNjEyMDV9.VBzPI5fnkDrfLisOmqxgXYUIid6m4r7QyAwNJM8xrKc"
+export const itsafact57 = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6Iml0c2FmYWN0NTdAZ21haWwuY29tIiwicHJlZmVyZWRMYW5nIjoiZnIiLCJpYXQiOjE1ODUzNjExNzF9.GvyrQOrq-7PgjYCuKgcfffABpvJqWz9dkMOArG565ZE"
+export const octopususer = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6Im9jdG9wdXN1c2VyQGFuZGVsYS5jb20iLCJwcmVmZXJlZExhbmciOm51bGwsImlhdCI6MTU4NTM2MTEyOX0.wcAWJWhp1SE1jnoKRrrTWPQ9tT-q7yzFLgyzIW-jYIM"
```
Comment (blaise82) [inline src/tests/mock/tokens.js]: please don't hard code tokens
just encode tokens using the function and the data you need in the token to test

**Review by blaise82 — CHANGES_REQUESTED** [review]

File: src/controllers/userController.js (hunk 3 above)
Comment (RusimbiPatrick) [reply src/controllers/userController.js]: sure

File: src/tests/mock/tokens.js (hunk 4 above)
Comment (RusimbiP) [reply src/tests/mock/tokens.js]: sure

File: src/controllers/userController.js (hunk 2 above)
Comment (RusimbiPatrick) [reply src/controllers/userController.js]: sure

File: src/tests/mock/tokens.js (hunk 4 above)
Comment (RusimbiPatrick) [reply src/tests/mock/tokens.js]: It slowed down the tests

Comment (blaise82) [reply src/tests/mock/tokens.js]: slowed down?

**Review by blaise82 — APPROVED** [review]

**Pull request merged**